File: pt5/1166.design-file-system.java
import java.util.*;

class FileSystem {
    private final Map<String, Integer> values;

    public FileSystem() {
        values = new HashMap<>();
        values.put("", -1);
    }

    public boolean createPath(String path, int value) {
        if (values.containsKey(path)) {
            return false;
        }

        String parent = path.substring(0, path.lastIndexOf('/'));
        if (!values.containsKey(parent)) {
            return false;
        }

        values.put(path, value);
        return true;
    }

    public int get(String path) {
        return values.getOrDefault(path, -1);
    }
}

File: pt5/1166.design-file-system.py
from __future__ import annotations


class FileSystem:
    def __init__(self):
        self.values = {"": -1}

    def createPath(self, path: str, value: int) -> bool:
        if path in self.values:
            return False

        parent = path.rsplit("/", 1)[0]
        if parent not in self.values:
            return False

        self.values[path] = value
        return True

    def get(self, path: str) -> int:
        return self.values.get(path, -1)

